org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/ConnectionStreamFactoryTest.java

```diff
@@ -45,6 +47,17 @@ public void testSocketSelection(){
 		System.clearProperty("STDOUT_PORT");
 	}
 
+	@Test
+	public void testStdInOut() throws IOException {
+		ConnectionStreamFactory tested = new ConnectionStreamFactory();
+		Assert.assertSame(tested.getInputStream(), JavaLanguageServerPlugin.getIn());
+		Assert.assertSame(tested.getOutputStream(), JavaLanguageServerPlugin.getOut());
+		Assert.assertNotSame(tested.getInputStream(), System.in);
+		Assert.assertNotSame(tested.getOutputStream(), System.out);
+		System.out.println("test");
+		Assert.assertTrue(tested.getInputStream().available() == 0);
```

Comment:
I guess `Assert.assertNotSame(tested.getOutputStream(), System.out);` is simpler :-)